### PR TITLE
feat: add optimus validate

### PR DESCRIPTION
**Pull request by irainia**
The changes on this PR consists of:

* adding job validate for Optimus, and
* marking job specification checks as deprecated

### CODE REVIEW

File: gotocompany/optimus/core/v1beta1/job_spec.proto [hunk 1]
```diff
@@ -104,12 +104,19 @@ service JobSpecificationService {
...
     };
   }
   // CheckJobSpecifications checks if the job specifications are valid
-  rpc CheckJobSpecifications(CheckJobSpecificationsRequest) returns (stream CheckJobSpecificationsResponse) {}
+  rpc CheckJobSpecifications(CheckJobSpecificationsRequest) returns (stream CheckJobSpecificationsResponse) {
+    option deprecated = true;
+  }
+
+  // Validate validates whether the job specifications are valid
+  rpc Validate(ValidateRequest) returns (ValidateResponse) {}
```
[the remark this answers is not in the document]
Comment (irainia) [reply gotocompany/optimus/core/v1beta1/job_spec.proto:119]: Actually I forgot to mention about this. Need your suggestion for the URL? 'cause I'm not sure a good one.

[the remark this answers is not in the document]
Comment (arinda-arif) [reply gotocompany/optimus/core/v1beta1/job_spec.proto:119]: how about `/v1beta1/project/{project_name}/namespace/{namespace_name}/job/validate` with POST method?
similar with JobInspect

[the remark this answers is not in the document]
Comment (irainia) [reply gotocompany/optimus/core/v1beta1/job_spec.proto:119]: sure, updating it

[the remark this answers is not in the document]
Comment (irainia) [reply gotocompany/optimus/core/v1beta1/job_spec.proto]: updated

**Review by arinda-arif — APPROVED** [review]

**Pull request merged**